src/libraries/Microsoft.PowerFx.Core/Public/IExpression.cs

```diff
@@ -10,10 +12,26 @@ namespace Microsoft.PowerFx.Core.Public
     /// </summary>
     /// <returns></returns>
     public interface IExpression
+    {
+        public Task<FormulaValue> EvalAsync(RecordValue parameters, CancellationToken cancel);

```

Comment:
>EvalAsync [](http://example.com/codeflow?start=34&length=9)

This is the key public facing change. add EvalAsync(), takes a Task and cancellation token.